torch/testing/_internal/common_methods_invocations.py

```diff
@@ -13803,7 +13803,7 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
            # vmap does not support random operations
            check_batched_forward_grad=False,
            dtypes=floating_types(),
```

Comment:
Can we also support half dtypes on CPU, so that CPU and CUDA support the same dtypes?

Reply:
I'm not that comfortable with writing  vectorized CPU code so I'd like to leave that for a later PR. There are a lot ops that do not have CPU fallbacks and floating types are already not supported on CPU for this op. @lezcano Would make a great future PR though.

Reply:
The changes needed to support float/bfloat16 on CPU are similar to the ones in CUDA, you won't need to write vectorised code.